Document global Redis/Valkey defaults for the platform (#973)
stacklok/docs-website: docs/platform/enterprise-platform/deployment.mdx

[Configure the Enterprise Cloud UI](../enterprise-cloud-ui/configure.mdx), and
[Configure the Registry Server](./configure-registry-server.mdx).

#### Global Redis/Valkey defaults

Several workloads can share an external Redis or [Valkey](https://valkey.io)
instance for distributed session storage. Rather than repeating the connection
on every workload, set it once under `global.redis` and let the components that
support it inherit the value. Valkey is a drop-in replacement for Redis and is
accepted wherever Redis is referenced. This block is optional: when
`global.redis.host` is empty, the global default is inactive and each workload
falls back to its own configuration.

```yaml title="values.yaml (global Redis/Valkey addition)"
global:
  redis:
    # When empty, the global default is inactive.
    host: 'redis.example.com'
    port: 6379
    # Set true to speak the Redis Cluster protocol to the host.
    clusterMode: false
    # Authentication uses a pre-created Secret, not plain-text values.
    # Create a Secret with the Redis password, then reference it here.
    existingSecret: 'valkey-auth'

>     existingSecretKey: 'redis-password'

Author: Good catch. Verified against the chart source: the operator only injects the password env vars when a Secret is set (`{{- if $rdSecret }}` in the operator `deployment.yaml`), and the AI Gateway guard only requires a host, never a Secret, so passwordless is supported. Fixed in fa2052f: the YAML comment and prose now state that auth is optional, `existingSecret` can be omitted for passwordless instances, and the Secret step only applies when authentication is required.

    tls:
      enabled: false
```

Plain-text password fields are intentionally absent: Helm values are stored
in-cluster and surfaced by `helm get values`, so credentials must come from a
Secret. Create it alongside the other secrets in [Step 2](#2-prepare-secrets):

> 

Author: Good catch. Verified against the chart source: the operator only injects the password env vars when a Secret is set (`{{- if $rdSecret }}` in the operator `deployment.yaml`), and the AI Gateway guard only requires a host, never a Secret, so passwordless is supported. Fixed in fa2052f: the YAML comment and prose now state that auth is optional, `existingSecret` can be omitted for passwordless instances, and the Secret step only applies when authentication is required.

[Configure the Enterprise Cloud UI](../enterprise-cloud-ui/configure.mdx), and
[Configure the Registry Server](./configure-registry-server.mdx).

#### Global Redis/Valkey defaults

Several workloads can share an external Redis or [Valkey](https://valkey.io)
instance for distributed session storage. Rather than repeating the connection
on every workload, set it once under `global.redis` and let the components that
support it inherit the value. Valkey is a drop-in replacement for Redis and is
accepted wherever Redis is referenced. This block is optional: when
`global.redis.host` is empty, the global default is inactive and each workload

> falls back to its own configuration.

Reviewer: ```suggestion
Several platform components can share an external Redis or Valkey instance for
distributed session storage. Rather than repeating the configuration on every
component, set it once under `global.redis` and let the components that support
it inherit the value. Valkey is a drop-in replacement for Redis. This block is
optional: when `global.redis.host` is empty, the global default is inactive and
each component falls back to its own configuration.
```

```yaml title="values.yaml (global Redis/Valkey addition)"
global:
  redis:
    # When empty, the global default is inactive.
    host: 'redis.example.com'
    port: 6379

>     # Set true to speak the Redis Cluster protocol to the host.

Reviewer: ```suggestion
    # Enable the Redis Cluster protocol when connecting to the host.
```

    clusterMode: false
    # Authentication is optional. Omit existingSecret for a passwordless
    # instance. When your Redis/Valkey requires a password, reference a

>     # pre-created Secret here; plain-text password fields are not supported.

Reviewer: This can be trimmed; lack of support for plaintext is implied by the secret references and expected by K8s admins.

```suggestion
    # Reference a pre-created Secret; omit for a passwordless instance.
```

    existingSecret: 'valkey-auth'
    existingSecretKey: 'redis-password'
    tls:
      enabled: false
```

If your Redis or Valkey instance is passwordless, leave `existingSecret` empty
and skip the rest of this step. When it requires authentication, the password
must come from a Secret rather than a plain-text value, because Helm values are
stored in-cluster and surfaced by `helm get values`. Create the Secret alongside

> the other secrets in [Step 2](#2-prepare-secrets):

Reviewer: Again, can trim the over-explanation here:

```suggestion
If passwordless, omit existingSecret and skip the Secret creation step. If your
instance requires authentication, create the Secret before installing the chart:
```

```

The following components inherit `global.redis`:

| Enable flag        | How it uses the global default                                                                                                                                                             |
| ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `toolhiveOperator` | Default session storage for `MCPServer`, `MCPRemoteProxy`, and `VirtualMCPServer` (vMCP) workloads that have no explicit `spec.sessionStorage`. The per-resource setting always overrides. |
| `aiGateway`        | Rate limiting and virtual API key storage, when the bundled Valkey instance is disabled.                                                                                                   |

The Enterprise Cloud UI, Enterprise Manager, and Registry Server do not use this
block. The embedded authorization server stores upstream tokens through the
`MCPExternalAuthConfig` resource, which is configured separately and is not
covered by the global default. See

> [Configure session storage](../../toolhive/guides-k8s/auth-k8s.mdx#configure-session-storage).

Reviewer: I don't think we need to list the components that don't need Redis at all; this could introduce the opposite confusion that they need Redis configured separately for some reason.

```suggestion
Note that the embedded auth server's token storage is configured separately via
`MCPExternalAuthConfig`. See
[Configure session storage](../../toolhive/guides-k8s/auth-k8s.mdx#configure-session-storage)
in the operator guide.
```

A question about that limitation; is it a gap we plan to close at some point so the embedded auth server also inherits the global config?
